rustls/src/key.rs

```diff
@@ -5,21 +5,88 @@ use std::fmt;
 /// The private key must be DER-encoded ASN.1 in either
 /// PKCS#8, PKCS#1, or Sec1 format.
 ///
-/// The `rustls-pemfile` crate can be used to extract
-/// private keys from a PEM file in these formats.
+/// A common format for storing private keys is
+/// [PEM](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail).
+/// PEM private keys are commonly stored in files with a `.pem` or `.key` suffix, and look like this:
+///
+/// ```txt
+/// -----BEGIN PRIVATE KEY-----
+///
```

Comment:
tiny nit: maybe there should be some indication there's content between the header/footer?
```suggestion
/// <encoded private key content>
```

Reply:
I think this is a good idea, might be even better to be explicit about it being base64-encoded.